components/widgets/forest-change/glad-alerts-daily/selectors.js

```diff
@@ -0,0 +1,204 @@
+import { createSelector, createStructuredSelector } from 'reselect';
+import moment from 'moment';
+import { format } from 'd3-format';
+import isEmpty from 'lodash/isEmpty';
+import sumBy from 'lodash/sumBy';
+import sortBy from 'lodash/sortBy';
+
+import { getChartConfig } from 'components/widgets/utils/data';
+
+const getAlerts = state => state.data?.data || null;
+const getColors = state => state.colors || null;
+const getStartDate = state => state.settings.startDate;
+const getEndDate = state => state.settings.endDate;
+const getSentences = state => state.sentences || null;
+const getLocationObject = state => state.location;
+const getOptionsSelected = state => state.optionsSelected;
+const getIndicator = state => state.indicator;
+const getStartIndex = state => state.settings.startIndex;
+const getEndIndex = state => state.settings.endIndex || null;
+
+const zeroFillDays = (startDate, endDate) => {
+  const start = moment(startDate);
+  const diffInDays = moment(endDate).diff(moment(startDate), 'days');
+  const dates = Array.from(Array(diffInDays).keys());
+
+  return [
+    startDate,
+    ...dates.map(() => start.add(1, 'days').format('YYYY-MM-DD'))
+  ];
+};
+
+export const getData = createSelector(
+  [getAlerts, getStartDate, getEndDate],
+  (data, startDate, endDate) => {
+    if (!data || isEmpty(data)) return null;
+
+    const zeroFilledData = zeroFillDays(startDate, endDate).map(date => ({
+      date,
+      alert__count: 0,
+      count: 0,
+      ...data.find(d => d.alert__date === date)
+    }));
+    return sortBy(zeroFilledData, 'date');
+  }
+);
+
+export const getStartEndIndexes = createSelector(
+  [getStartIndex, getEndIndex, getData],
+  (startIndex, endIndex, currentData) => {
+    if (!currentData) {
+      return {
+        startIndex,
+        endIndex
+      };
+    }
+
+    const start =
+      startIndex || startIndex === 0 ? startIndex : currentData.length - 7;
```

Comment:
any reason for 7? I think for this case we can make it 365 for the last year as it covers many years

Reply:
Fixed!